### ISSUE TITLE
Add multi-zone support

### ISSUE THREAD

**Issue by rytilahti** [rytilahti/python-songpal#13]
Some devices offer possibility to have multiple zones with different inputs and outputs (including separate volume controls etc.), support for that could be added.

Relevant discussion and some logs can be found also in https://github.com/rytilahti/python-songpal/issues/8 .

**Comment by little-boots** [comment]
I have a STR-DN1080 (multi-zone receiver), but I'm still very new to Python and don't think I could efficiently update this code myself.  Is there something else I could provide (logs, etc.) that would be helpful?

**Comment by rytilahti** [comment]
At the moment there is not much to do. However, if you like, you could generate a devinfo dump (`songpal dump_devinfo > STR-DN1080.json` or similar)  and create a PR to add it into this repository for future reference.

Another thing that could be helpful is to describe what type of interface should be provided for those other zones? What commands are wanted?

**Comment by little-boots** [comment]
I tried creating a fork/pull request with the dump.  I'm hoping I did it right?  I rarely use Github, so please let me know if I messed something up.

In any case, I normally interact with songpal via Home Assistant.  Nearly, all commands work perfectly in HA, but it always reports the source as 'Main Zone', rather than the actual source that zone is set to.  It also lists the zones together with the sources.  Instead, it should give the option to select zone, then source within that zone.

Not sure if it helps, but this is the output of "songpal inputs":
```
  * Main Zone (uri: extOutput:zone?zone=1) (active)
  * Zone 2 (uri: extOutput:zone?zone=2)
  * HDMI Zone (uri: extOutput:zone?zone=4)
  * SOURCE (uri: extInput:source)
    - extOutput:zone?zone=2
    - extOutput:zone?zone=4
  * BD/DVD (uri: extInput:bd-dvd)
    - extOutput:zone?zone=1
    - extOutput:zone?zone=4
  * BYOD (uri: extInput:game)
    - extOutput:zone?zone=1
    - extOutput:zone?zone=4
  * SAT/CATV (uri: extInput:sat-catv)
    - extOutput:zone?zone=1
    - extOutput:zone?zone=2
    - extOutput:zone?zone=4
  * Fire (uri: extInput:video?port=1)
    - extOutput:zone?zone=1
    - extOutput:zone?zone=2
  * Chrome (uri: extInput:video?port=2)
    - extOutput:zone?zone=1
    - extOutput:zone?zone=4
  * TV (uri: extInput:tv)
    - extOutput:zone?zone=1
  * Roku (uri: extInput:sacd-cd) (active)
    - extOutput:zone?zone=1
    - extOutput:zone?zone=2
    - extOutput:zone?zone=4
  * Bluetooth Audio (uri: extInput:btAudio)
    - extOutput:zone?zone=1
    - extOutput:zone?zone=2
```

Finally, songpal floods the HA logs with warnings (several per minute) which I can't seem to suppress:
`The device seems to have more than one volume setting.`

**Comment by rytilahti** [comment]
The warning has been fixed already, and the fix has been merged to home-assistant so it will be fixed in the next release (https://github.com/home-assistant/home-assistant/pull/16297).

As there is currently no zone logic in the media_player component, it's a bit tricky to implement this. One possibility would be to have one media_player per each zone, but there are probably several small issues to pull this off "blindly" without a device, basically requiring someone, who has such a device, to implement it.

**Comment by little-boots** [comment]
Thanks - that will be great if the warnings are fixed!  I tried uploading the dump again; just let me know if there are any more issues.

I also forgot that you're bound by how Home Assistant's components are set up...  Having a separate media_player for each zone would definitely provide a complete solution.  But for my purposes, it would be enough to simply report whatever input is active within the Main Zone as the "Source", rather than reporting which zone is active when multiple zones are discovered.  I only ever use Zone 1, so seeing what's playing within that zone is much more important to me. Not sure if that's easier for you or if it works for others, though.

**Comment by jwiese** [comment]
I just added my device info as well for STR-DN1060, and submitted a PR. Hopefully that is helpful.

I think a media_player for each zone is probably the right thing here. I'm happy to contribute to this if I could get a push in the right direction.

**Comment by rytilahti** [comment]
If you want to work on this, the way would be to find what is the best place to list the available zones (and how to get its available inputs/sources). As the volume control is related to that output, the code needs probably a bit refactoring to attach the volume controller information to that. I think it will make sense to create a Zone class containing/binding all the relevant logic, this can also be used then for single-zone devices as well.

**Comment by jwiese** [comment]
FYI, I've started working on this. Hope to have more updates soon!

**Comment by Otitori** [comment]
> FYI, I've started working on this. Hope to have more updates soon!

Thanks jwiese, looking forward to your updates 👍

**Comment by jwiese** [comment]
#42 Is a first stab at zone control stuff. Works well for me, would love to hear if it works for other peoples' setups

**Comment by rytilahti** [comment]
Fixed with #42 thanks to @jwiese, please open new issues if you encounter any troubles with it!
